packages/loaders/src/zarr/pixel-source.ts

```diff
@@ -101,37 +111,48 @@ class ZarrPixelSource<S extends string[]> implements PixelSource<S> {
     // and ignored by deck.gl.
     if (xStart === xStop || yStart === yStop) {
       throw new BoundsCheckError('Tile slice is zero-sized.');
+    } else if (xStart < 0 || yStart < 0 || xStop > width || yStop > height) {
+      throw new BoundsCheckError('Tile slice is out of bounds.');
     }
+
     return [slice(xStart, xStop), slice(yStart, yStop)];
   }
 
-  async getRaster({ selection }: RasterSelection<S> | { selection: number[] }) {
-    const sel = this._chunkIndex(selection, null, null);
-    const { data, shape } = (await this._data.getRaw(sel)) as RawArray;
-    const [height, width] = shape;
+  private async _getRaw(
+    selection: (null | Slice | number)[],
+    getOptions?: { storageOptions?: any }
+  ) {
+    // @ts-expect-error - TODO: fix ZarrJS types, this is correct
+    const result = await this._data.getRaw(selection, getOptions);
```

Comment:
We should just use `getRaw`. If the requests align with the chunking, than great. This PR in zarr.js will allow for passing abort signal (but this will also be supported in zarrita): https://github.com/gzuidhof/zarr.js/pull/146#pullrequestreview-1651753041

Reply:
Updated in v0.6.2 of ZarrJS